### ISSUE TITLE
Feature Request: Support Extras / Special Features

### ISSUE THREAD

**Issue by R1CH-T** [tinyMediaManager/tinyMediaManager#79]
__What TMM version are you using?__
2.8.3.1

__release, pre-release, nightly, or directly from GitHub/branch?__
Release

__What is the actual behaviour?__
tMM does not scrape contents of "Extras" sub-directory.

__What is the expected behaviour?__
It would be nice if it did. ;)

__Steps to reproduce:__
- first
- second
- ...
N/A: Feature request.

__Additional__
Have you attached the Logfile from the day it happened?
Your Operating system? (win/mac/linux? version?)
N/A

I have been working on [getting Kodi's _VideoExtras_ addon working](http://forum.kodi.tv/showthread.php?tid=214289&pid=2408939#pid2408939) on the new Krypton beta version's Estuary skin and in my research, randomly stumbled upon the site [http://dvdspecialfeatures.net/](url).

This reminded me that I have been meaning to raise this issue with you folks for a while and never quite got around to it.

I'm not sure if the site mentioned above would be of any use for scraping extras, but that being said (and I don't want to waste your time), this would be a killer feature.

This site might be a one-off and of no practical use in this context, but I have in the past noticed some _special feature_ or _extras_ type stuff listed in TMDB in the past; most notably some of "The Matrix Collection" documentaries, which were successfully scraped.

Please can you look into this, so maybe the ability to search and scrape the contents of an "Extras" subdirectory can be added to tMM.

Thanks,
Rich.

**Comment by myron0815** [comment]
Extras should work as designed;
Just put every video file into extra/extras folder (or follow the dedicated plex folder naming)
do NOT create other folders there.

Since we're talking about video files, i'm quite sure, such extras aren't public available... so we cannot scrape them....?

**Comment by myron0815** [comment]
after reading again - you want to scrape information for extras?
This is not supported; and there is place to store information (for Kodi)
And we cannot map "just an extras name.avi" with an online extra information.....

**Comment by R1CH-T** [comment]
Sorry, I just checked my inbox an saw that you'd replied to this query.

While you are correct that this is not supported in Kodi directly, neither are the extras. The VideoExtras addon is required to access and view these video files.

NFO files for extras are supported within the VideoExtras addon and use the same naming convention and structure as Kodi does for the main movie. These files currently have to be manually created and written for each of the extras contained within the Extras folder, as detailed in the wiki entry for the addon: http://kodi.wiki/view/Add-on:VideoExtras#NFO_Files

While, at the moment, there is not a dedicated database containing information about extras for DVDs and Blurays out there, I have seen a few sites popping up recently on the web and TMDB does have some of these in it's database. I discovered this by mistake, because I had accidentally placed some extras for "The Matrix Ultimate Collection" in the wrong place and when I "searched and scraped" the collection with tMM, one of these extras (a documentary) was listed in the search results dialog and allowed me to download all of the relevant metadata etc. for this documentary and imported it into tMM.

This is when I thought about the possibility of adding a "Search and Scrape Movie Extras" feature into tMM.

It might end up being a laborious process to scrape maybe hundreds of movie extras using tMM and most people would probably be fine with keeping their extras listed by filename. However, if people out there (like me) would like to be able to list these by their proper name and in the right order, with artwork etc., having the ability to use tMM to help automate this process would be a real time-saver.

Manually creating these metadata files would be a total nightmare and is just th problem tMM was created to solve.

I hope this helps to clarify my original query and thank you for taking the time to read this.

Rich
